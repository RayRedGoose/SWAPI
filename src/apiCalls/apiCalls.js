export const getMoviesData = () => {
  return fetch('https://swapi.co/api/films/')
    .then(response => {
      if (!response.ok) {
        throw Error('Movies wasn\'t fetched')
      }
      return response.json()
    })
    .then(data => data.results.sort((a, b) => {
      return a.episode_id - b.episode_id
    }))
}

export const getAnyData = (url, type) => {
  return fetch(url)
    .then(response => {
      if (!response.ok) {
        throw Error(`${type} wasn\'t fetched`)
      }
      return response.json()
    })
}

// const getCharacterData = (array) => {
//   return array.map((character, index) => {
//     return fetch(character)
//     .then(response => response.json())
//   })
// }
//
// const makePromises = Promise.all(getCharacterData)
//   .then(promises => promises.map(promise => {
//
//     const characterName = promise.name;
//
//     return Promise.all([speciesFetch(promise), planetFetch(promise), filmsFetch(promise)])
//     .then(info => {return {name: characterName, species: info[0], planet: info[1].planetName, population: info[1].planetPopulation, films: info[2]}})
//     .then(characterStats => {this.setState({characters: [...this.state.characters, characterStats]})})
//   }))
//
//   const speciesFetch = (promise) => {
//     fetch(promise.species)
//     .then(res => res.json())
//     .then(data => data.name)
//   }
//
//   const planetFetch = (promise) => {
//     fetch(promise.homeworld)
//     .then(res => res.json())
//     .then(data => {return {planetName: data.name, planetPopulation: data.population}})
//   }
//
//   const filmsFetch = (promise) => {
//     const filmsPromises = promise.films.map(film => {
//       return fetch(film)
//       .then(res => res.json())
//       .then(data => data.title)
//
//       return Promise.all(filmsPromises)
//       .then(data => {return data})
//     })
//   }
