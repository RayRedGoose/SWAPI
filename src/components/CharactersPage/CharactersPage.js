import React, { Component } from 'react';
import UserProfile from '../UserProfile/UserProfile';
import CharacterBox from '../CharacterBox/CharacterBox';
import { Route } from 'react-router-dom';
import './CharactersPage.scss';
import Header from '../Header/Header'

export default class CharactersPage extends Component {
  constructor() {
    super()
  }

  componentDidMount = () => {
    this.props.addMovies()
  }

  render() {
    return !this.props.movie ? null: (
      <section className="characters-page">
        <Header
          user={this.props.user}
          logOut={this.props.logOut}
          header={`Episode ${this.props.movie.episode_id}: {this.props.movie.title}`} />
        <main className="crawl-text">
          <section>
            <div>
              <h4>{this.props.movie.title}</h4>
              <p>{this.props.movie.opening_crawl}</p>
            </div>
          </section>
        </main>
        <CharacterBox
          characters={this.props.movie.characters}
          addMovies={this.props.addMovies}
          addFavorite={this.props.addFavorite}
          removeFavorite={this.props.removeFavorite}
        />

      </section>
    )
  }
}
