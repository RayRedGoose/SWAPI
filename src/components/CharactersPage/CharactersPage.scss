.charactersPage {
  min-width: 100vw; min-height: 100vh;
  position: relative;
  h2 {
    font-size: 3.2em;
  }
}

.scroll-box {
  width: 90%;
  height: 40%;
}
