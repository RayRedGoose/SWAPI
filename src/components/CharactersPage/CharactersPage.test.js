import React from 'react'
import CharactersPage from './CharactersPage'
import movieResults from '../../mockData/fakeMovies.js'
import { getMoviesData } from '../../apiCalls/apiCalls'
import { shallow } from 'enzyme'

describe('CharactersPage', () => {
  let wrapper
  const addMovies = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <CharactersPage
        id='2'
        movie={{
          title: 'The Phantom Menace',
          release_date: '1999-05-19',
          episode_id: 1,
          opening_crawl: 'Turmoil has engulfed the, Galactic Republic.',
          characters: ['C-3PO']
        }}
        user={{
          name: 'Taylor Swift',
          quote: "I find your lack of faith disturbing",
          rank: 'intermediate'}}
        logOut={jest.fn()}
        favorites={[]}
        addFavorite={jest.fn()}
        removeFavorite={jest.fn()}
      />
    )
  })

  it('should match snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
