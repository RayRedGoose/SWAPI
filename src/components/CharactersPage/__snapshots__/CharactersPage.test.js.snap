// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersPage should match snapshot if no movie data was passed 1`] = `""`;

exports[`CharactersPage should match snapshot with all data passed in correctly 1`] = `
<section
  className="characters-page"
>
  <Header
    headerText="Episode 1: the fast and the furious"
    isButton={true}
    logOut={[MockFunction]}
    user={
      Object {
        "name": "Taylor Swift",
        "quote": "I find your lack of faith disturbing",
        "rank": "intermediate",
      }
    }
  />
  <main
    className="crawl-text"
  >
    <section>
      <div>
        <h4>
          the fast and the furious
        </h4>
        <p>
          paaaam-pam paaaaam-pam pa-pa-paaaam
        </p>
      </div>
    </section>
  </main>
  <div>
    <h3>
      All Movie Characters
    </h3>
    <FavoriteFilter
      number={0}
    />
  </div>
  <CharacterBox
    addFavorite={[MockFunction]}
    addMovies={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    characters={
      Array [
        Object {
          "films": Array [
            "bambi",
            "little women",
            "sound of music",
          ],
          "isFavorite": false,
          "name": "Luke",
          "planet": "Tatioone",
          "population": 20000,
          "species": "human",
        },
        Object {
          "films": Array [
            "die hard",
          ],
          "isFavorite": true,
          "name": "Leia",
          "planet": "Alderaan",
          "population": 230000,
          "species": "human",
        },
        Object {
          "films": Array [
            "lawrence of arabia",
          ],
          "isFavorite": false,
          "name": "Obi-wan",
          "planet": "Whereever he is from",
          "population": 40000,
          "species": "human",
        },
        Object {
          "films": Array [
            "punch drunk love",
          ],
          "isFavorite": true,
          "name": "Darth Vadar",
          "planet": "Tatioone",
          "population": 50000,
          "species": "robot",
        },
        Object {
          "films": Array [
            "cinderella",
          ],
          "isFavorite": false,
          "name": "Han",
          "planet": "Millenium Falcon",
          "population": 60000,
          "species": "human",
        },
      ]
    }
    favorites={Array []}
    removeFavorite={[MockFunction]}
  />
</section>
`;
